google-http-client/src/main/java/com/google/api/client/http/javanet/NetHttpTransport.java

```diff
@@ -118,11 +123,21 @@ public NetHttpTransport() {
   NetHttpTransport(ConnectionFactory connectionFactory,
       SSLSocketFactory sslSocketFactory, HostnameVerifier hostnameVerifier) {
     this.connectionFactory =
-        connectionFactory == null ? new DefaultConnectionFactory() : connectionFactory;
+        getConnectionFactory(connectionFactory);
     this.sslSocketFactory = sslSocketFactory;
     this.hostnameVerifier = hostnameVerifier;
   }
 
+  private ConnectionFactory getConnectionFactory(ConnectionFactory connectionFactory) {
+    if (connectionFactory == null) {
+      if (System.getProperties().getProperty("com.api.client.should_use_proxy") != null) {
```

Comment:
com.google.api.client.should_use_proxy

Reply:
This is design question I was not sure about: should there be a flag to enable proxies on the SDK, or should it always pick it up if the properties are set?

Reply:
I would say use a flag… @garrettjonesgoogle thoughts?